src/formio/components/AddressNL.js

```diff
@@ -349,3 +401,42 @@ const PostCodeField = ({required}) => {
     />
   );
 };
+
+const HouseNumberField = ({required, allowDeriveAddress}) => {
+  const {values, setFieldValue, getFieldProps} = useFormikContext();
+  const {onBlur: onBlurFormik} = getFieldProps('houseNumber');
+  const {baseUrl} = useContext(ConfigContext);
+  const postcode = values['postcode'];
+  const houseNumber = values['houseNumber'];
+
+  const fetchStreetNameAndCity = async () => {
```

Comment:
this code is now duplicated, you can extract it into a single function used by both fields, and you might even just pass it in as a prop from the parent component:

```jsx
const {values, setFieldValue} = useFormikContext();

const autofillAddress = () => {
    if (!allowDeriveAddress) return;
    if (!values.postcode) || (!values.houseNumber)} return;
    
    const data = await get(`${baseUrl}geo/address-autocomplete`, params);
    setFieldValue('city', data['city']);
    setFieldValue('streetName', data['streetName']);
    setFieldValue('secretStreetCity', data['secretStreetCity']);
};

// ...

return (
  <>
    <PostCodeField required={required} autoFillAddress={autofillAddress} />
  </>
);
```

and then your onBlur becomes:

```jsx
const onBlur = (event) => {
  onBlurFormik(event);
  autoFillAddress();
};
```